lib/stages/visRawReader.cpp

```diff
@@ -296,5 +308,15 @@ void visRawReader::main_thread() {
         frame_id = (frame_id + 1) % out_buf->num_frames;
         read_ind++;
         ind++;
+
+        // Get the end time for the loop and sleep for long enough to satisfy
+        // the max rate
+        end_time = current_time();
+        double sleep_time = min_read_time - (end_time - start_time);
+        DEBUG("Sleep time {}", sleep_time);
+        if (sleep_time > 0) {
+            auto ts = double_to_ts(sleep_time);
+            nanosleep(&ts, NULL);
```

Comment:
I guess the sleep times end up so short, that `usleep` wouldn't be precise enough here?

Reply:
Nah, I don't think they're that short, but as it was easy enough for me to get a timespec I figured why not.